bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/PowerMaxBinding.java

```diff
@@ -0,0 +1,881 @@
+ *
+ * @author lolodomo
+ * @since 1.9.0
+ */
+public class PowerMaxBinding extends AbstractActiveBinding<PowerMaxBindingProvider>
+        implements ManagedService, PowerMaxEventListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(PowerMaxBinding.class);
+
+    /** Default delay in milliseconds to reset a motion detection */
+    private static final int DEFAULT_MOTION_OFF_DELAY = 3 * 60000;
+
+    /** The serial port to use for connecting to the PowerMax alarm system */
+    private String serialPort = null;
+
+    /** The IP address and TCP port to use for connecting to the PowerMax alarm system */
+    private String ipAddress = null;
+    private int tcpPort = 0;
+
+    /** The delay in milliseconds to reset a motion detection */
+    private int motionOffDelay = DEFAULT_MOTION_OFF_DELAY;
+
+    /** The PIN code to use for arming/disarming the PowerMax alarm system from openHAB */
+    private String pinCode = null;
+
+    /** Enable or disable arming the PowerMax alarm system from openHAB */
+    private boolean allowArming = false;
+
+    /** Enable or disable disarming the PowerMax alarm system from openHAB */
+    private boolean allowDisarming = false;
+
+    /** Force the standard mode rather than trying using the Powerlink mode */
+    private boolean forceStandardMode = false;
+
+    /** Panel type used when in standard mode */
+    private PowerMaxPanelType panelType = PowerMaxPanelType.POWERMAX_PRO;
+
+    /** Automatic sync time */
+    private boolean autoSyncTime = false;
+
+    /** The object to store the current state of the PowerMax alarm system */
+    private PowerMaxState currentState = null;
+
+    /** Number of items managed by the binding */
+    private int itemCount = 0;
+
+    /** Boolean indicating whether or not the update of items is temporarily disabled by the binding */
+    private boolean itemUpdateDisabled = true;
+
+    /**
+     * Boolean indicating whether or not an update of all items will be required
+     * during the next refresh cycle
+     */
+    private boolean triggerItemsUpdate = false;
+
+    /** Boolean indicating whether or not connection is established with the alarm system */
+    private boolean connected = false;
+
+    public PowerMaxBinding() {
+    }
+
+    /**
+     * Activates the binding. Actually does nothing, because on activation
+     * OpenHAB always calls updated to indicate that the config is updated
+     * Activation is done there
+     */
+    @Override
+    public void activate() {
+        logger.debug("Activate PowerMax alarm binding");
+    }
+
+    /**
+     * Deactivates the binding
+     */
+    @Override
+    public void deactivate() {
+        logger.debug("Deactivate PowerMax alarm binding");
+        closeConnection();
+    }
+
+    /**
+     * @{inheritDoc
+     */
+    @Override
+    protected long getRefreshInterval() {
+        return 20000; // 20 seconds
```

Comment:
Any reason to make this configurable with a `refresh` property in the .cfg file?